### DISCUSSION TITLE
Performance:  Running gap on Windows vs running on linux implementation on same hardware

### DISCUSSION THREAD

**Discussion by JoeMooreBrk** [gap-system/gap#5228] [Q&A]
This may be a naïve question, but I'll ask it anyway.  If I'm running on current but medium performance x86 hardware, should I run gap as configured for Windows or set up a linux machine on the same hardware.  All of my tools are currently on the Windows side, but I'm open to change if the performance gains are large enough.  If they are only incremental, it's not worth it except for the education value.

Background--I am a very experienced Windows programmer who is fascinated with basic Group Theory but most of my context in my real world has not needed a great deal of performance.

Thanks.

**Comment by ChrisJefferson** [comment]
This raised an interesting answer, as the answer is different to what I thought it was (and I'm going to try and investigate why that is the case).

Short answer: One of the easiest ways to get the "best of both worlds" is to install the Windows Subsystem for Linux (look for "ubuntu" in the windows app store). This will install a well-integrated Linux VM on your windows machine, and in that the performance is (on my computer) actually faster than booting into "true" linux (I think because Windows does better at managing my machine's CPU / heat production).

The "windows installer" version of GAP has two issues:

1) The time to read files is very slow -- so GAP can take around 10 seconds to start instead of 1 second on Linux. Once calculations are being performed however, this largely goes away.

2) I thought the time taken for CPU-intensive calculations was about the same (it was last time I measured it, many years ago), but I'm actually getting a consistent 2x slowdown and I don't know what's causing that!

**Reply by jaslamMiktex** [reply]
Actually, I have not experienced any slowdown on the WSL/Linux GAP.   I also happen to run a very computationally intensive application.
If you are not using a very complex application, and it is only for trial runs, then you would be happy to try the HPC GAP version. It has some hiccups on the way, but eventually, you can make it run, and use ALL the cores on your CPU.
I have achieved a factor of 3 performance improvement. Of course, it will also depend very much on the inherent parallelism in the application.
There are a few examples of multiprocessing in the HPC GAP package.
